packages/react-server/src/ReactFlightServer.js

```diff
@@ -2705,11 +2705,18 @@ function emitErrorChunk(
   if (__DEV__) {
     let message;
     let stack = '';
+    let env = request.environmentName;
     try {
       if (error instanceof Error) {
         // eslint-disable-next-line react-internal/safe-string-coercion
         message = String(error.message);
         stack = getStack(error);
+        const errorEnv = (error: any).environmentName;
+        if (typeof errorEnv === 'string') {
+          // This probably came from another FlightClient as a pass through.
```

Comment:
```suggestion
          // This probably came from another FlightServer as a pass through.
```

Reply:
Well, technically what I mean is that the FlightClient is what creates the object on the parsing side. In theory it wasn't necessarily even from a FlightServer that generated the payload.

Conceptually it's a bit unclear who hides the original thrown/rejected value (which might not be an Error) and who creates the new Error - FlightServer or FlightClient? I guess FlightServer. We don't technically serialize Error objects - we serialize the `throw`.

Reply:
>Well, technically what I mean is that the FlightClient is what creates the object on the parsing side.

Do you mean [this line](https://github.com/facebook/react/pull/29846/files#diff-4bc3e2f66cfadc63bd7abcc0387fcc7c1373092f832cac1b34614d8cddca5eacR1621)? How would this end up back in a Flight Server?

Reply:
In the server-to-server case where an RSC server consumes the payload from another RSC server (not the binary pass-though case). I.e. the case we have in our tests.

Reply:
[oh, of course](https://github.com/facebook/react/blob/fb3b9ead472ec0254e4ce7bd9000eb100010f08e/packages/react-client/src/__tests__/ReactFlight-test.js#L2338). 🤦